### PR TITLE
Prepare v0.6.0

### PR DESCRIPTION
**Pull request by behackl**
## Changelog
- Added generated changelog
- Bumped version to v0.6.0

### CODE REVIEW

**Comment by behackl** [comment]
I would still like to get #1430 and #1431 in. Please everyone @ManimCommunity/core check out the generated changelog (as soon as RTD finished the build)!

File: docs/source/changelog/0.6.0-changelog.rst
Comment (AntonBallmaier) [inline docs/source/changelog/0.6.0-changelog.rst:18]:
```suggestion
* Anton Ballmaier
```

I have authored a patch in 0.5 as well. Other names may be wrong as well?

File: docs/source/changelog/0.6.0-changelog.rst
Comment (AntonBallmaier) [inline docs/source/changelog/0.6.0-changelog.rst:210]:
```suggestion
* `#1283 <https://github.com/ManimCommunity/manim/pull/1283>`__: Set default value of keyword ``random_seed`` in Scene to ``None`` (was 0 and fixed before)
```

File: docs/source/changelog/0.6.0-changelog.rst
Comment (AntonBallmaier) [inline docs/source/changelog/0.6.0-changelog.rst:220]:
```suggestion
* `#1249 <https://github.com/ManimCommunity/manim/pull/1249>`__: Set corners of :class:`~.Rectangle` in counterclockwise direction to be consistent with other geometry `Mobject`s. `Transform`ations between `Rectangle` and other geometry `Mobject`s look better now.
```

Comment (AntonBallmaier) [reply docs/source/changelog/0.6.0-changelog.rst:220]: It sound a bit repetitive.

**Review by AntonBallmaier — CHANGES_REQUESTED** [review]

File: docs/source/changelog/0.6.0-changelog.rst
Comment (jsonvillanueva) [reply docs/source/changelog/0.6.0-changelog.rst:18]: Can you link the PR, this may help to debug the dev_changelog.py script.

File: docs/source/changelog/0.6.0-changelog.rst
Comment (jsonvillanueva) [reply docs/source/changelog/0.6.0-changelog.rst:220]: I modified the body of this: https://github.com/ManimCommunity/manim/pull/1249
@abhi-6988 @abhi3019  Please review and modify if you find the language undescriptive.

File: docs/source/changelog/0.6.0-changelog.rst
Comment (jsonvillanueva) [reply docs/source/changelog/0.6.0-changelog.rst:210]: Fixed the title and also added a reference to :class:`~.Scene`

File: docs/source/changelog/0.6.0-changelog.rst
Comment (jsonvillanueva) [reply docs/source/changelog/0.6.0-changelog.rst:18]: Nvm, I found it: 
image

The issue here is that your name changed between v0.4.0 -> v0.5.0 and v0.5.0 -> v0.6.0  from `AntonBallmaier` to `Anton Ballmaier` with a space -- our script works by pulling the author name and comparing author names from the set of past contributors to current contributors. I suppose we could do it based on a combination of email/user

File: docs/source/changelog/0.6.0-changelog.rst
Comment (abhi3019) [reply docs/source/changelog/0.6.0-changelog.rst:220]: > I modified the body of this: #1249

Looks good, no issues with that thanks! I've also updated the title and changelog of #1385, please have a look

File: docs/source/changelog/0.6.0-changelog.rst
Comment (abhi3019) [inline docs/source/changelog/0.6.0-changelog.rst:157]: CC @Darylgolden 
```suggestion
   - Added :meth:`~.Mobject.match_points`, which transforms the points, positions and submobjects of a Mobject to match that of the other while keeping style unchanged.
```

**Comment by behackl** [comment]
I fixed some more broken formatting and fixed a few links; I'm more or less fine with the current state of this. I'll be away for a few hours, everyone with write access can re-run the script though (or push changes to the changelog directly).

File: docs/source/changelog/0.6.0-changelog.rst
Comment (Darylgolden) [reply docs/source/changelog/0.6.0-changelog.rst]: LGTM

**Comment by Darylgolden** [comment]
Also, can I ask how the list of contributors who reviewed PRs is determined? I tried to review several PRs this release, even though I don't have write access.

**Comment by jsonvillanueva** [comment]
> Also, can I ask how the list of contributors who reviewed PRs is determined? I tried to review several PRs this release, even though I don't have write access.

https://github.com/ManimCommunity/manim/blob/master/scripts/dev_changelog.py#L92

If you go to the `Files changed` tab there's a green dropdown button that allows you to make Reviews. I don't think comments are included in the GitHub API request.

**Comment by Darylgolden** [comment]
> > Also, can I ask how the list of contributors who reviewed PRs is determined? I tried to review several PRs this release, even though I don't have write access.
> 
> https://github.com/ManimCommunity/manim/blob/master/scripts/dev_changelog.py#L92
> 
> If you go to the `Files changed` tab there's a green dropdown button that allows you to make Reviews. I don't think comments are included in the GitHub API request.

I'm pretty sure I did that several times.

**Comment by jsonvillanueva** [comment]
> I'm pretty sure I did that several times.

I'm not really sure what the issue is. We're not doing any filtering on whether the review is of type `approves/request changes/comments` and you do appear in at least one REST API query: https://api.github.com/repos/ManimCommunity/manim/pulls/1347/reviews

**Comment by Darylgolden** [comment]
I initially thought the issue was write access, but there are people without write access in the reviewers list too 😕

**Comment by Darylgolden** [comment]
I also think @hydrobeam reviewed some PRs but isn't included too, such as at #1430

**Comment by kolibril13** [comment]
Thanks for making the pr, I just saw that my name is also not there in the "have been reviewed" section.

**Comment by behackl** [comment]
For those not in the reviewer list: can you check whether you have a name set in your github profile? If not, then that's the issue. I think we can fix the situation by using the login name if no actual name is set.

**Comment by kolibril13** [comment]
> For those not in the reviewer list: can you check whether you have a name set in your github profile? If not, then that's the issue. I think we can fix the situation by using the login name if no actual name is set.

I did not set my name, but it is set now :)

**Comment by hydrobeam** [comment]
I've also set my name 🚀

File: docs/source/changelog/0.6.0-changelog.rst
Comment (naveen521kk) [reply docs/source/changelog/0.6.0-changelog.rst]: can you change it in the PR title, then behackl would render this again?

Comment (behackl) [reply docs/source/changelog/0.6.0-changelog.rst]: I've already taken care of it. 👍

File: docs/source/changelog/0.6.0-changelog.rst
Comment (naveen521kk) [inline docs/source/changelog/0.6.0-changelog.rst:20]: I think both the names are the same right? @cobordism

**Comment by Darylgolden** [comment]
I've set my (fake) name

**Comment by behackl** [comment]
PRs #1434 (obviously) as well as #1282 #1200 #1259 were not yet included in the list because their merge message have either been modified, or because they were merged manually/differently. I've added them now manually.

Also, I've changed the logic of how authors are determined -- it now also uses the GitHub API, the names in the authors and reviewers list should now be consistent.

File: docs/source/changelog/0.6.0-changelog.rst
Comment (jsonvillanueva) [inline docs/source/changelog/0.6.0-changelog.rst:124]: This indentation needs to be fixed, but I think this s3ection is honestly pretty redundant:
```suggestion

```

**Review by huguesdevimeux — APPROVED** [review]

File: docs/source/changelog/0.6.0-changelog.rst
Comment (jsonvillanueva) [inline docs/source/changelog/0.6.0-changelog.rst:85]:
```suggestion
   :class`~.VectorField` is renamed to :class:`~.ArrowVectorField` and a new class :class:`~.VectorField` is added as a superclass for :class:`~.ArrowVectorField` and :class:`~.StreamLines`. `AnimatedStreamLines` is removed. It's functionality is moved to :class:`~.StreamLines`. Added a lot of new options when working with vector fields. :class:`~.ShowPassingFlashWithThinningStrokeWidth` was moved to the indication module.
```

**Review by jsonvillanueva — APPROVED** [review]
I'm approving this in advance with these changes (Gotta make the first PR look great and remove the malformatted bullets); however, I'd like to mention that there could be a more consistent grammar/authoring of this text. As is, it appears to be written by various authors and in some cases there's redundancies between the PR title/body text. We'd be here all day doing nitpicks and the meaning is understood nonetheless.

**Review by kolibril13 — APPROVED** [review]

**Comment by behackl** [comment]
I won't regenerate the changelog at this point, I think it is good to go. (Certainly not written uniformly, and some information might be redundant -- but I don't think there is a wide margin for improvement here.)

Pretty impressive work everyone, and thanks for reviewing the changelog! I'll merge and then release soon™️, if you have further comments or remarks, it's either now or never.

**Review by AntonBallmaier — APPROVED** [review]

**Review by kilacoda — APPROVED** [review]

**Pull request merged**
